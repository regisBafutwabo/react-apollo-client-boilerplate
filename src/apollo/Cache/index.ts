import { TypePolicy } from "apollo/TypePolicies";
import { persistCache } from "apollo3-cache-persist";

import { InMemoryCache } from "@apollo/client";

async function initCache(): Promise<InMemoryCache> {
    const cache: InMemoryCache = new InMemoryCache({
        typePolicies: TypePolicy,
    });

    await persistCache({
        cache,
        storage: window.localStorage,
        debug: process.env.NODE_ENV === "development",
    });

    return cache;
}
export default initCache;
