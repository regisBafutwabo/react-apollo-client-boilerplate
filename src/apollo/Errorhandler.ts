import { onError } from "@apollo/client/link/error";

export const errorLink = onError(({ graphQLErrors, networkError }) => {
    if (graphQLErrors) {
        graphQLErrors.map(({ message, locations, path }) => {
            console.error(
                `[GraphQL error]: Message: ${message}, Location: ${locations}, Path:${path}`,
            );
            return null;
        });
    }
    if (networkError) {
        console.error(`[Network error]: ${networkError}`);
    }
});
