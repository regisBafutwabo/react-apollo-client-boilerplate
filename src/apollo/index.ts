import { InMemoryCache, ApolloClient } from "@apollo/client";
import { typeDefs } from "./TypeDefs";
import { LocalResolvers } from "./Operations/Client/Mutations";
import link from "./Middlewares";
import initCache from "./Cache";
/**
 * If you are not using no graphql connection connection
 */
let client: ApolloClient<any>;

export const getApolloClient = async (): Promise<ApolloClient<any>> => {
    if (client) return client;

    const cache: InMemoryCache = await initCache();

    const apolloClient: ApolloClient<any> = new ApolloClient({
        link,
        cache,
        connectToDevTools: process.env.NODE_ENV === "development",
        typeDefs,
        resolvers: LocalResolvers,
    });

    return apolloClient;
};
