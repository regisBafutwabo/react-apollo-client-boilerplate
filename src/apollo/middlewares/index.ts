import { ApolloLink } from "@apollo/client";
import serverLink from "./server";
import { errorLink } from "./errors";

const link: ApolloLink = ApolloLink.from([serverLink, errorLink]);

export default link;
