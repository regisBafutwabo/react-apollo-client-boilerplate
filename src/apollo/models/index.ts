export * from "./i18n.model";
export * from "./todo.model";
