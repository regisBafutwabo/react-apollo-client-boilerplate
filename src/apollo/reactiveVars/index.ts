export * from "./todo";
export * from "./i18n";
