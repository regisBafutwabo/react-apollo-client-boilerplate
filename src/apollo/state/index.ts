export * from "./i18n.initial";
export * from "./user.initial";
