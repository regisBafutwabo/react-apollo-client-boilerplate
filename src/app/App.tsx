import React from "react";
import { Switch, Route, Redirect, RouteComponentProps } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import theme from "styles/theme";
import { ErrorPage, Welcome } from "./loadable";

const App = () => {
    return (
        <ThemeProvider theme={theme}>
            <Switch>
                <Route exact path="/" component={Welcome} />
                <Route
                    exact
                    path="/NotFound"
                    render={(props: RouteComponentProps<any>) => (
                        <ErrorPage type="404" {...props} />
                    )}
                />
                <Route
                    exact
                    path="/Unauthorized"
                    render={(props: RouteComponentProps<any>) => (
                        <ErrorPage type="403" {...props} />
                    )}
                />
                <Route
                    exact
                    path="/Error"
                    render={(props: RouteComponentProps<any>) => (
                        <ErrorPage type="500" {...props} />
                    )}
                />

                <Redirect to="/NotFound" />
            </Switch>
        </ThemeProvider>
    );
};

export default App;
