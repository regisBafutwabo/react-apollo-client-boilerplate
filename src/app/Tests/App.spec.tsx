import React from "react";

import { getApolloClient } from "apollo";
import createI18n from "i18n/i18n";
import { I18nextProvider } from "react-i18next";

import { ApolloProvider } from "@apollo/client";
import { render } from "@testing-library/react";

import App from "../index";

describe("App Component", () => {
    test("renders App Component", async () => {
        const client = await getApolloClient();
        const i18n = await createI18n(client);

        const { getByText } = render(
            <ApolloProvider client={client}>
                <I18nextProvider i18n={i18n}>
                    <App />
                </I18nextProvider>
            </ApolloProvider>,
        );

        const linkElement = getByText(/Loading.../i) || getByText(/Welcome to React/i);
        expect(linkElement).toBeInTheDocument();
    });
});
