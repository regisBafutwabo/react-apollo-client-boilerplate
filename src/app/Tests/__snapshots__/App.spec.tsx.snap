// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match with the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
